### PR TITLE
feat(i18n): allow mustache syntax in translations

### PR DESCRIPTION
**Pull request by hypeJunction**
Translations now support {{mustache}} syntax.

### CODE REVIEW

**Comment by hypeJunction** [comment]
Original concept was borrowed from @brettp in #9833, but instead of reinventing the wheel, we just use existing mustache engines.

File: engine/classes/Elgg/I18n/Translator.php [hunk 1]
```diff
@@ -142,33 +159,63 @@ public function translate($message_key, array $args = [], $language = "") {
 		$langs['en'] = true;
 
 		// try to translate
-		$notice = '';
 		$string = $message_key;
+
+		// separate into sprintf_args and kwargs
+		list($printf_args, $kwargs) = self::getStringReplacementsArgs($args);
```
Comment (hypeJunction) [inline engine/classes/Elgg/I18n/Translator.php]: We don't want to encourage object method usage, so we should just flatten `$kwargs` here using `ElggData::toObject()`

**Comment by hypeJunction** [comment]
Fixes #6908

**Comment by hypeJunction** [comment]
@juho-jaakkola Do we need to do anything to allow this in Transifex?

**Comment by juho-jaakkola** [comment]
On a quick look, I didn't find any mention whether Transifex supports this syntax in PHP arrays.

**Comment by hypeJunction** [comment]
Can add during 3.x

**Pull request closed without merging**